src/components/[guild]/RolePlatforms/components/EditRewardAvailabilityModal/components/StartEndTimeForm.tsx

```diff
@@ -0,0 +1,99 @@
+import { FormControl, FormLabel, Input, Stack, Text } from "@chakra-ui/react"
+import { Control, Path, useFormContext, useWatch } from "react-hook-form"
+import { PlatformName } from "types"
+import { DAY_IN_MS } from "utils/formatRelativeTimeFromNow"
+
+type Props<TFieldValues, TContext> = {
+  // Passing control, so we can infer the proper field names for the other 2 props
+  control: Control<TFieldValues, TContext>
+  startTimeField: Path<TFieldValues>
+  endTimeField: Path<TFieldValues>
+  platformType: PlatformName
+}
+
+export const AUTO_TIMEFRAME_PLATFORMS: PlatformName[] = ["POAP"]
+
+export const getShortDate = (isoDate: string): string | undefined => {
+  if (!isoDate) return undefined
+  return isoDate.split(/:\d{2}\.\d{3}Z/)[0]
+}
+
+export const datetimeLocalToIsoString = (datetimeLocal: string): string | null => {
+  if (!datetimeLocal) return null
+
+  try {
+    return new Date(datetimeLocal).toISOString()
+  } catch {
+    return null
+  }
+}
+
+const StartEndTimeForm = <TFieldValues, TContext>({
```

Comment:
Mostly copy-pasted it from `EditRewardAvailabilityModal`, just added generics to it, so we can use it in a typesafe way everywhere.